contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,441 @@
+        @param amount_ Amount of sOHM debt issued from donor to recipient
+        @param recipient_ Address to direct staking yield and vault shares to
+     */
+    function depositSohm(uint256 amount_, address recipient_) external returns (uint256 depositId) {
+        require(isValidDeposit(amount_, recipient_), "Not a valid deposit.");
+
+        IERC20(sOHM).safeTransferFrom(msg.sender, address(this), amount_);
+        IERC20(sOHM).approve(address(staking), amount_);
+        uint256 gohmAmount = staking.wrap(address(this), amount_);
+
+        depositId = _deposit(msg.sender, gohmAmount);
+        recipientIds[recipient_].push(depositId);
+        recipientLookup[depositId] = recipient_;
+
+        emit Deposited(msg.sender, recipient_, gohmAmount);
+    }
+
+    /**
+        @notice Deposit additional gOHM, and update deposit record
+        @param id_ Deposit ID to direct additional gOHM to
+        @param amount_ Amount of new gOHM debt issued from donor to recipient
+     */
+    function addToDeposit(uint256 id_, uint256 amount_) external {
+        require(isValidUpdate(id_, amount_), "Not a valid deposit update.");
+        require(depositInfo[id_].depositor == msg.sender, "Can't adjust deposits that aren't yours");
+
+        IERC20(gOHM).safeTransferFrom(msg.sender, address(this), amount_);
+        _addToDeposit(id_, amount_);
+
+        emit DepositUpdated(msg.sender, recipientLookup[id_], amount_);
+    }
+
+    /**
+        @notice Deposit additional sOHM, wrap to gOHM, and update deposit record
+        @param id_ Deposit ID to direct additional gOHM to
+        @param amount_ Amount of new sOHM debt issued from donor to recipient
+     */
+    function addToSohmDeposit(uint256 id_, uint256 amount_) external {
+        require(isValidUpdate(id_, amount_), "Not a valid deposit update.");
+        require(depositInfo[id_].depositor == msg.sender, "Can't adjust deposits that aren't yours");
+
+        IERC20(sOHM).safeTransferFrom(msg.sender, address(this), amount_);
+        IERC20(sOHM).approve(address(staking), amount_);
+        uint256 gohmAmount = staking.wrap(address(this), amount_);
+        _addToDeposit(id_, gohmAmount);
+
+        emit DepositUpdated(msg.sender, recipientLookup[id_], gohmAmount);
+    }
+
+    /**
+        @notice Withdraw donor's gOHM from vault and subtracts debt from recipient
+        @param id_ Deposit ID to remove gOHM debt from
+        @param amount_ Amount of gOHM debt to remove and return to donor
+     */
+    function withdrawPrincipal(uint256 id_, uint256 amount_) external {
+        require(isValidWithdrawal(amount_), "Not a valid withdrawal.");
+        DepositInfo storage currDeposit = depositInfo[id_];
+        require(currDeposit.depositor == msg.sender, "Can't withdraw principal that's not yours");
+
+        if (amount_ >= IgOHM(gOHM).balanceTo(currDeposit.principalAmount)) {
+            _withdrawPrincipal(id_, amount_);
```

Comment:
this will revert right? 
here were meant to make amount_ = IgOHM(gOHM).balanceTo(currDeposit.principalAmount)
or set a new variable called amountToWithdraw = IgOHM(gOHM).balanceTo(currDeposit.principalAmount).

Reply:
doesn't revert, but probably cleaner to do it your way and not in-line. will change

Reply:
done

Reply:
oh. sorry, misinterpreted that. Yes, it will revert if the user tries to withdraw the max amount I believe. I'll test now. I think that probably having it instead default to withdrawing the max amount if they request more is the better way to go about it. Thoughts @ind-igo?